### PR TITLE
feat: web vitals in onboarding

### PR DESCRIPTION
**Pull request by pauldambra**
Screenshot 2024-09-23 at 11 28 06

We've added web vitals autocapture, but someone has to go looking to find it
and yet still the majority have turned it on

so, let's enable it by default and let people opt out

### CODE REVIEW

**Comment by pauldambra** [comment]
@PostHog/team-growth #lazyweb

is there a way to auto-create the web vitals dashboard already if folk have web vitals enabled after this step

(just being lazy and asking rather than trying to figure it out :))

**Review by daibhin — APPROVED** [review]

**Review by marandaneto — APPROVED** [review]

**Review by zlwaterfield — APPROVED** [review]
RE: create dashboard, @raquelmsmith has been working with dashboards so she will be the best person to ask.

File: frontend/src/scenes/onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -134,6 +134,14 @@ const ProductAnalyticsOnboarding = (): JSX.Element => {
...
             visible: true,
         },
+        {
+            title: 'Enable web vitals autocapture',
+            description: `Uses Google's web vitals library to automagically capture performance information.`,
+            teamProperty: 'autocapture_web_vitals_opt_in',
+            value: currentTeam?.autocapture_web_vitals_opt_in ?? true,
+            type: 'toggle',
+            visible: true,
+        },
```
Comment (raquelmsmith) [inline frontend/src/scenes/onboarding/Onboarding.tsx:144]: Do we want to add this to replay, flags, and surveys as well? So people can turn it on regardless of what product they are using.

File: frontend/src/scenes/onboarding/Onboarding.tsx
Comment (raquelmsmith) [inline frontend/src/scenes/onboarding/Onboarding.tsx:1]: @pauldambra 

> is there a way to auto-create the web vitals dashboard already if folk have web vitals enabled after this step

There isn't a built-in way to do this right now, but you can def do it!

On the `OnboardingStep` `continueAction` (which gets called on continue) you can check if they have it enabled, and if so then use this func to create the dashboard

https://github.com/PostHog/posthog/blob/8c4939d7105d0939f050e7456d7c4f63f6383181/frontend/src/scenes/dashboard/newDashboardLogic.ts#L79-L89

Don't redirect after creation, no vars (I think that template uses no vars.. maybe check)

File: frontend/src/scenes/onboarding/Onboarding.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/onboarding/Onboarding.tsx:144]: 🤔 feels weird to me to have cross-product toggles... but then we have session replay in the product analytics one

i might leave things as they are for now

File: frontend/src/scenes/onboarding/Onboarding.tsx
Comment (pauldambra) [reply frontend/src/scenes/onboarding/Onboarding.tsx:1]: Ah, you need to API call to load templates 🤔 
Will add to my to-do list and come back to this when I can focus 🙈

**Pull request merged**